src/main/java/com/mysema/maven/apt/AbstractProcessorMojo.java

```diff
@@ -341,13 +384,21 @@ public void execute() throws MojoExecutionException {
             }
             ExecutorService executor = Executors.newSingleThreadExecutor();
             try {
-                CompilationTask task = compiler.getTask(out, fileManager, null, compilerOptions, null, compilationUnits1);
+                // the access to buildContext has to made from the current thread (uses ThreadLocal)
+                final List<Diagnostic<? extends JavaFileObject>> diagnostics = new ArrayList<Diagnostic<? extends JavaFileObject>>();
+                CompilationTask task = compiler.getTask(out, fileManager, new DiagnosticListener<JavaFileObject>() {
+                    @Override
+                    public void report(Diagnostic<? extends JavaFileObject> diagnostic) {
+                        diagnostics.add(diagnostic);

```

Comment:
Could you instead of making a new anonymous class use the `DiagnosticCollector` that exists in the same package?
It provides `getDiagnostics()` which you can feed directly in your current method `processDiagnostics`, no changes necessary.
It is also thread safe, for good measure, and it fails fast. so now in that nullcheck I commented on you can be extra sure it is non-null.